src/Compilers/CSharp/Test/Symbol/Symbols/SymbolEqualityTests.cs

```diff
@@ -744,7 +743,202 @@ public class B
+
+            VerifyEquality(method1ContainingType, method2ContainingType,
+                expectedDefault: true,
+                expectedIncludeNullability: false
+                );
+        }
+
+        [Fact]
+        [WorkItem(8195, "https://github.com/dotnet/roslyn/issues/38195")]
+        public void SemanticModel_SubstitutedEvent_Equality()
+        {
+            var source =
+@"
+#nullable enable
+public class A<T> 
+    where T : class //not neccesary, but makes it easier to reason about the resulting fields
+{
+    public event System.EventHandler<T> MyEvent;
+
+    public static void M(A<T> t)
+    {
+        _  = t.MyEvent;
+    }
+}
+";
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics(
+                // (6,41): warning CS8618: Non-nullable event 'MyEvent' is uninitialized. Consider declaring the event as nullable.
+                //     public event System.EventHandler<T> MyEvent;
+                Diagnostic(ErrorCode.WRN_UninitializedNonNullableField, "MyEvent").WithArguments("event", "MyEvent").WithLocation(6, 41)
+                );
+
+            var syntaxTree = comp.SyntaxTrees[0];
+            var root = syntaxTree.GetRoot();
+
+            var member1Syntax = (ClassDeclarationSyntax)root.DescendantNodes().First(sn => sn.Kind() == SyntaxKind.ClassDeclaration);
+            var member2Syntax = (IdentifierNameSyntax)root.DescendantNodes().Last(sn => sn.Kind() == SyntaxKind.IdentifierName);
+
+            var model = comp.GetSemanticModel(syntaxTree);
+
+            var event1 = (IEventSymbol)((INamedTypeSymbol)model.GetDeclaredSymbol(member1Syntax)).GetMembers("MyEvent").Single(); // A<T!>! A<T>.M(A<T!>! t)
+            var event2 = (IEventSymbol)model.GetSymbolInfo(member2Syntax).Symbol;                                                 // A<T!>! A<T!>.M(A<T!>! t)
+
+            Assert.IsType(typeof(SourceFieldLikeEventSymbol), event1);
+            Assert.IsType(typeof(SubstitutedEventSymbol), event2);
+
+            VerifyEquality(event1, event2,
+                expectedDefault: true,
+                expectedIncludeNullability: false,
+                expectHashCodeMatch: false // different symbols
+                );
+
+            var event1Type = event1.Type; // A<T!>
+            var event2Type = event2.Type; // A<T!>
+
+            VerifyEquality(event1Type, event2Type,
+                expectedDefault: true,
+                expectedIncludeNullability: true
+                );
+
+            var event1ContainingType = event1.ContainingType; //A<T>
+            var event2ContainingType = event2.ContainingType; //A<T!>
+
+            Assert.IsType(typeof(SourceNamedTypeSymbol), event1ContainingType);
+            Assert.IsType(typeof(ConstructedNamedTypeSymbol), event2ContainingType);
+
+            VerifyEquality(event1ContainingType, event2ContainingType,
+                expectedDefault: true,
+                expectedIncludeNullability: false
+                );
+        }
```

Comment:
>} [](start = 8, length = 1)

It looks like we are lacking tests for generic methods, matrix [(not constructed, constructed) method] * [(not constructed, constructed) containing type].